src/renderers/dom/shared/flattenStyleArray.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule flattenStyleArray
+ * @flow
+ */
+
+'use strict';
+
+type NestedStylesArray = Array<Object | NestedStylesArray | null>;
+
+function flattenStyleArray(nestedStylesArray: NestedStylesArray): Object {
+  var result = {};
+  var nestedStylesCopy = nestedStylesArray.slice();
+
+  var i = 0;
+  while (i < nestedStylesCopy.length) {
+    var styleObjOrArray = nestedStylesCopy[i];
+    if (Array.isArray(styleObjOrArray)) {
+      nestedStylesCopy.splice(i, 1, ...styleObjOrArray);
+      // correct for the fact that the nested items are now at position i
+      i--;
+    } else if (typeof styleObjOrArray !== 'undefined'
```

Comment:
FYI, the reason we are doing `typeof X !== 'undefined'` is because if you do `X !== undefined` and X isn't a global variable, JS is going to throw an exception.

In this case, `styleObjOrArray` is a local variable so it always exists in scope and you can safely do `styleObjOrArray !== undefined`.